RELEASE.md

```diff
@@ -1,39 +1,37 @@
-- Regression and Feature Verification Automated tests.
+- Regression and Feature Verification Automated tests
 - Exploratory testing by QA engineers
 - Strict security scanners on the container images
 - Upgrade from previous releases
-- Beta testing by users on issues that they are interested in.
-- Dogfooding on OpenEBS workload and e2e infrastructure clusters.
+- Beta testing by users on issues that they are interested in
 
-If any issues are found during the above stages, they are fixed and a new release candidate builds are generated.
+If any issues are found during the above stages, they are fixed and the prerelease tag is overridden by the newer changes and are up for above action items again.
 
-Once all the above tests are completed, a main release tagged image is published.
+Once all the above tests are completed, a main release is created.
 
 ## Release Tagging
 
-zfs-localpv is released as container image with versioned tag.
+LocalPV ZFS is released with container images and a respective helm chart as the only recommended way of installation. Even though the [zfs-operator](./deploy/zfs-operator.yaml) is also published, it is generated by templating the helm chart itself.
 
-Before creating a release, the repo owner needs to create a separate branch from the active branch, which is `develop`. Name of the branch should follow the naming convention of `v.0.7.x` if release is for 0.7.0.
+Before creating a release, the repo owner needs to create a separate branch from the active branch, which is `develop`. Name of the branch should follow the naming convention of `release/2.7` if release is for `2.7.x`.
 
-Once the release branch is created, changelog from `changelogs/unreleased` needs to be moved to release specific folder, if release branch is `v0.7.x` then folder will be `changelogs/v0.7.x`.
+Upon creation of a release branch ex. `release/2.7`, two automated PRs open up to change the chart versions of the charts in `release/2.7` branch to `2.7.0-prerelease` and `develop` to `2.8.0-develop`. Post merge of these two PRs, the `2.7.0-prerelease` and `2.8.0-develop` tags are pushed to respective docker registries and also the respective helm charts against these tags are published. The prerelease versions increment via automated PRs on every release creation. For example once `2.7.0` is published a `2.7.1-prerelease` image and chart would be published to allow testing of further patch releases and so on.
 
-The format of the release tag is either "Release-Name-RC1" or "Release-Name" depending on whether the tag is a release candidate or a release. (Example: 0.7.0-RC1 is a GitHub release tag for zfs-localpv release candidate build. 0.7.0 is the release tag that is created after the release criteria are satisfied by the zfs-localpv builds.)
+The format of the release tag follows semver versioning. The final release tag is of format `X.Y.Z` and the respective prerelease and develop tags are `X.Y.Z-prerelease` and `X.Y+1.0-develop`.
 
```

Comment:
It should be verified if all significant changes to the project have been itemized on to the CHANGELOG.md file, before a release is cut.

@Abhinandan-Purkait this is my take on the changelog update process

Reply:
Done.